### PR TITLE
Feature/applying filters

### PR DESCRIPTION
**Pull request by ghost**
Apply filters on todos

### CODE REVIEW

File: frontend/src/components/Filters.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export default function Filters({
+  selectedFilter,
+  categories,
+  changeFilter,
+}: filtersProps) {
+  return (
+    <ul className="filter-list">
+      {categories.map((item: "ALL" | "PENDING" | "COMPLETED") => {
+        return selectedFilter === item ? (
+          <li
```
Comment (alinauman02) [inline frontend/src/components/Filters.tsx]: Like mentioned in the meeting, a `li` is not an element that a user can interact with. While adding a `onClick` property does work, it is not semantically correct.

Use a more appropriate element inside a `li`, like a `button`

File: frontend/src/components/Filters.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export type Category = "ALL" | "PENDING" | "COMPLETED";
+
+interface filtersProps {
```
Comment (alinauman02) [inline frontend/src/components/Filters.tsx]: Why is this name in `camelCase` instead of `PascalCase`?

File: frontend/src/App.css [hunk 3]
```diff
@@ -93,9 +93,35 @@ body {
   background-color: #0a2ea4;
 }
 
+.check-box {
+  transform: scale(1.5);
+  margin-right: 5%;
+}
 .todo-form {
```
Comment (alinauman02) [inline frontend/src/App.css:101]: Shouldn't there be a space between these two different styles? The same for all other styles here.

**Review by alinauman02 — CHANGES_REQUESTED** [review]

File: frontend/src/App.css [hunk 4]
```diff
@@ -93,9 +93,42 @@ body {
...
+
 .todo-form {
   display: flex;
   width: 200%;
   vertical-align: top;
   text-align: center;
 }
+
+.category-items {
+  float: left;
```
Comment (alinauman02) [inline frontend/src/App.css]: We no longer use float to define a layout in CSS. If you need to arrange some items in a row or a column, use flexbox instead. You can use [this](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) guide to learn about flex box.

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -1,18 +1,40 @@
...
 import { Todo } from "./models/todo.model";
+import Filters, { Category } from "./components/Filters";
 
 const urlString = "http://localhost:3001/todos/";
 
 function App() {
   const [todos, setTodos] = useState<Todo[]>([]);
   const [newTodoDescription, setNewTodoDescription] = useState("");
   const [loading, setLoading] = useState(true);
+  const [selectedFilter, setSelectedFilter] = useState<Category>("ALL");
```
Comment (ZunairaT) [inline frontend/src/App.tsx]: define an enum for categories rather than using strings like `ALL`

File: frontend/src/components/Filters.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export default function Filters({
+  selectedFilter,
+  categories,
+  changeFilter,
+}: FilterProps) {
+  return (
+    <ul className="filter-list">
+      {categories.map((item: "ALL" | "PENDING" | "COMPLETED") => {
```
Comment (ZunairaT) [inline frontend/src/components/Filters.tsx]: why not reuse the type Category you defined at the top of the file

**Review by ZunairaT — CHANGES_REQUESTED** [review]

**Review by alinauman02 — DISMISSED** [review]
Good job!

**Review by ZunairaT — DISMISSED** [review]

**Comment by alinauman02** [comment]
@Ejazhussainbaltistani There is a conflict in the PR. Resolve this and then you can merge.

**Review by alinauman02 — APPROVED** [review]

**Pull request merged**